kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/Log4j2DivertAppender.scala

```diff
@@ -75,11 +75,12 @@ class Log4j2DivertAppender(
    */
   override def append(event: LogEvent): Unit = {
     super.append(event)
-    // That should've gone into our writer. Notify the LogContext.
-    val logOutput = writer.toString
-    writer.reset()
-    val log = OperationLog.getCurrentOperationLog
-    if (log != null) log.write(logOutput)
+    writer.synchronized {
```

Comment:
this is the key change

Reply:
not sure, whether it will have performance issue.
 and `synchronized` is not supported to interrupt.

Reply:
`synchronized` is cheap in sync mode.

Reply:
Does not the error occur before this line?

Reply:
> Does not the error occur before this line?

The error occurs in reading because it does not use read-write-lock properly in writing.

In superclass `AbstractWriterAppender#append`, it uses `readWriteLock` for concurrency control, we are actually touching the write buffer, so `writeLock` is required here.

`synchronized(writer)` is a cheaper(I guess) implementation, but I moved to use `writeLock` here to make the semantics clearer.